File: array_problems/countMostCommonChar/countMostCommonChar.js
class StringProblems {
    constructor(str) {
        this.str = str;
    }

    countMostCommonChar() {
        this.str = this.str.replace(/[^a-zA-Z]/g, "").toLowerCase();
        let charCounts = {};
        for (let i = 0; i < this.str.length; i++) {
            charCounts[this.str.charAt(i)] = (charCounts[this.str.charAt(i)] || 0) + 1;
        }
        let maxCount = 0;
        let maxChar = '';
        for (let char in charCounts) {
            if (charCounts[char] > maxCount) {
                maxCount = charCounts[char];
                maxChar = char;
            }
        }
        return maxChar;
    }
}

const stringProblems = new StringProblems("Hello world, this is very basic for developers");
console.log(stringProblems.countMostCommonChar());

File: array_problems/countMostCommonChar/countMostCommonChar.cs
using System;
using System.Threading.Tasks;
using System.Collections.Generic;
using System.Linq;
using System.Text.RegularExpressions;

class ArrayProblems {
    public async Task<char> CountMostCommonChar(string str) {
        str = Regex.Replace(str, "[^a-zA-Z]", "").ToLower();
        Dictionary<char, int> Obj = new Dictionary<char, int>();
        foreach (char ch in str) {
            if (Obj.ContainsKey(ch)) {
                Obj[ch]++;
            } else {
                Obj[ch] = 1;
            }
        }
        char maxKey = Obj.FirstOrDefault(x => x.Value == Obj.Values.Max()).Key;
        return maxKey;
    }
}
public class Program {
    public static async Task Main(string[] args) {
        ArrayProblems arrayProblems = new ArrayProblems();
        char mostCommonChar = await arrayProblems.CountMostCommonChar("Hello world, this is very basic for developers");
        Console.WriteLine(mostCommonChar);
    }
}